File: sdk/go/data_governance.go
package llmengine

import (
	"context"
	"fmt"
)

type DataGovernanceClient struct {
	client *Client
}

type DataAsset struct {
	AssetID        string `json:"asset_id"`
	TenantID       string `json:"tenant_id"`
	Name           string `json:"name"`
	Type           string `json:"type"`
	Classification string `json:"classification"`
}

func (c *DataGovernanceClient) ListAssets(ctx context.Context, tenantID string) ([]DataAsset, error) {
	var resp struct {
		Assets []DataAsset `json:"assets"`
	}
	err := c.client.get(ctx, fmt.Sprintf("/v1/data-governance/assets?tenant_id=%s", tenantID), &resp)
	if err != nil {
		return nil, err
	}
	return resp.Assets, nil
}

File: sdk/typescript/src/data-governance.ts
export interface DataAsset {
  tenant_id: string;
  asset_id: string;
  name: string;
  type: string;
  classification: string;
  domain: string;
  status: string;
}

export interface DataAccessRequest {
  tenant_id?: string;
  principal_id: string;
  asset_id: string;
  action?: string;
  purpose?: string;
}

export class DataGovernanceClient {
  private baseUrl: string;

  constructor(baseUrl: string) {
    this.baseUrl = baseUrl;
  }

  async listAssets(tenantId: string = "global"): Promise<{ assets: DataAsset[] }> {
    const res = await fetch(`${this.baseUrl}/v1/data-governance/assets?tenant_id=${tenantId}`);
    return res.json();
  }

  async evaluateAccess(req: DataAccessRequest): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/data-governance/access/evaluate`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(req),
    });
    return res.json();
  }

  async getTrustScore(assetId: string, tenantId: string = "global"): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/data-governance/trust/${assetId}?tenant_id=${tenantId}`);
    return res.json();
  }
}
